src/routes/Search.js

```diff
@@ -2,27 +2,33 @@ import React from 'react';
 import { RouteHandler } from 'react-router';
 import axios from 'axios';
 import FlexView from 'react-flexview';
+import LoadingSpinner from '../components/LoadingSpinner';
 import SearchResultPanel from 'components/Search/SearchResultPanel';
 import EmptyResult from 'components/Search/EmptyResult';
 import UndefinedResult from 'components/Search/UndefinedResult';
 import ErrorResult from 'components/Search/ErrorResult';
 
 export default class Search extends React.Component {
 
-  state = { repositories: null, errorMsg: null };
+  state = { repositories: null, errorMsg: null, isFetching: false };
 
   assignRespositories = (response) => {
-    this.setState({ repositories: response.data.items, errorMsg: null });
+    this.setState({ repositories: response.data.items, errorMsg: null, isFetching: false });
   }
 
   errorHandler = () => {
-    this.setState({ repositories: null, errorMsg: 'Search failed!' });
+    this.setState({ repositories: null, errorMsg: 'Search failed!', isFetching: false });
   }
 
   fetchRepositories = (q) => {
-    axios.get('http://api.github.com/search/repositories', { params: { q } })
-    .then(this.assignRespositories)
-    .catch(this.errorHandler); // eslint-disable-line
+    //reset the states
+    this.setState({ repositories: null, errorMsg: null, isFetching: false });
+    if (typeof(q) !== 'undefined' && q !== null) {
```

Comment:
in the correct PR to handle the empty search, I used it correctly.

Also using typeof in this format is correct:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof